lib/rspec/core/formatters/helpers.rb

```diff
@@ -70,8 +70,7 @@ def self.format_seconds(float, precision=nil)
         # @param string [String] string with trailing zeros
         # @return [String] string with trailing zeros removed
         def self.strip_trailing_zeroes(string)
-          stripped = string.sub(/[^1-9]+$/, '')
-          stripped.empty? ? "0" : stripped
+          string.sub(/(?:(\..*[^0])0+|\.0+)$/, '\1')
```

Comment:
Can you add a rubular explanation here saying what this is doing e.g. http://rubular.com/r/ujWack5Utu